perception/autoware_compare_map_segmentation/src/lanelet_elevation_filter/node.cpp

```diff
@@ -0,0 +1,245 @@
+
+  // Initialize timing utilities following fusion node pattern
+  stop_watch_ptr_ = std::make_unique<autoware_utils::StopWatch<std::chrono::milliseconds>>();
+
+  // Initialize debug publisher if enabled
+  if (params_.enable_debug) {
+    try {
+      // Initialize DebugPublisher following fusion node pattern
+      debug_publisher_ = std::make_unique<autoware_utils::DebugPublisher>(this, get_name());
+
+      // Initialize debug markers publisher
+      debug_markers_pub_ = this->create_publisher<visualization_msgs::msg::MarkerArray>(
+        "~/debug/elevation_grid_markers", rclcpp::QoS(1).transient_local());
+
+      RCLCPP_INFO(this->get_logger(), "Debug publishers initialized");
+    } catch (const std::exception & e) {
+      RCLCPP_WARN(this->get_logger(), "Failed to initialize debug publishers: %s", e.what());
+      params_.enable_debug = false;  // Disable debug mode if initialization fails
+    }
+  }
+
+  // Subscribe to lanelet map
+  map_sub_ = this->create_subscription<autoware_map_msgs::msg::LaneletMapBin>(
+    "~/input/lanelet_map", rclcpp::QoS(1).transient_local(),
+    std::bind(&LaneletElevationFilterComponent::onMap, this, std::placeholders::_1));
+
+  RCLCPP_INFO(this->get_logger(), "LaneletElevationFilter has been initialized.");
+
+  printParameters();
+}
+
+void LaneletElevationFilterComponent::printParameters()
+{
+  RCLCPP_INFO(this->get_logger(), "=== Lanelet Elevation Filter Configuration ===");
+  RCLCPP_INFO(this->get_logger(), "Grid resolution: %.2f m", params_.grid_resolution);
+  RCLCPP_INFO(this->get_logger(), "Height threshold: %.2f m", params_.height_threshold);
+  RCLCPP_INFO(this->get_logger(), "Sampling distance: %.2f m", params_.sampling_distance);
+  RCLCPP_INFO(this->get_logger(), "Target frame: %s", params_.target_frame.c_str());
+  RCLCPP_INFO(this->get_logger(), "Debug mode: %s", params_.enable_debug ? "enabled" : "disabled");
+}
+
+void LaneletElevationFilterComponent::loadParameters()
+{
+  // Load parameters with default values
+  params_.grid_resolution = this->declare_parameter<double>("grid_resolution", 1.0);
+  params_.height_threshold = this->declare_parameter<double>("height_threshold", 2.0);
+  params_.sampling_distance = this->declare_parameter<double>("sampling_distance", 0.5);
+  params_.target_frame = this->declare_parameter<std::string>("target_frame", "map");
+  params_.enable_debug = this->declare_parameter<bool>("enable_debug", false);
+
+  // Validate parameters
+  if (params_.grid_resolution <= 0.0) {
+    throw std::invalid_argument("grid_resolution must be positive");
+  }
+  if (params_.height_threshold <= 0.0) {
+    throw std::invalid_argument("height_threshold must be positive");
+  }
+  if (params_.sampling_distance <= 0.0) {
+    throw std::invalid_argument("sampling_distance must be positive");
+  }
+}
+
+void LaneletElevationFilterComponent::filter(
+  const PointCloud2ConstPtr & input, [[maybe_unused]] const IndicesPtr & indices,
+  PointCloud2 & output)
```

Comment:
@kaancolak Which `frame_id` are you supposed to processing the pointcloud? 
IMO, it is better to documentation it?